iceoryx_utils/include/iceoryx_utils/internal/relocatable_pointer/pointer_repository.hpp

```diff
@@ -38,142 +42,72 @@ class PointerRepository
-            {
-                m_info[id].basePtr = nullptr;
-
-                // do not search for next lower registered index but we could do it here
-                return true;
-            }
-        }
-
-        return false;
-    }
-
-    void unregisterAll()
-    {
-        for (auto& info : m_info)
-        {
-            info.basePtr = nullptr;
-        }
-        m_maxRegistered = 0u;
-    }
-
-    ptr_t getBasePtr(id_t id)
-    {
-        if (id <= MAX_ID && id >= MIN_ID)
-        {
-            return m_info[id].basePtr;
-        }
-
-        // for id 0 nullptr is returned, meaning we will later interpret a relative pointer
-        // by casting the offset into a pointer (i.e. we measure relative to 0)
-
-        return nullptr; // we cannot distinguish between not registered and nullptr registered, but we do not need to
-    }
-
-    id_t searchId(ptr_t ptr)
-    {
-        for (id_t id = 1u; id <= m_maxRegistered; ++id)
-        {
-            // return first id where the ptr is in the corresponding interval
-            if (ptr >= m_info[id].basePtr && ptr <= m_info[id].endPtr)
-            {
-                return id;
-            }
-        }
-        // implicitly interpret the pointer as a regular pointer if not found
-        // by setting id to 0
-        // rationale: test cases work without registered shared memory and require
-        // this at the moment to avoid fundamental changes
-        return 0u;
-        // return INVALID_ID;
-    }
-
-    bool isValid(id_t id)
-    {
-        return id != INVALID_ID;
-    }
-
-    void print()
-    {
-        for (id_t id = 0; id < m_info.size(); ++id)
-        {
-            auto ptr = m_info[id].basePtr;
-            if (ptr != nullptr)
-            {
-                std::cout << id << " ---> " << ptr << std::endl;
-            }
-        }
-    }
+    /// @brief default constructor
+    PointerRepository() noexcept;
+
+    /// @brief registers the start pointer of the segment in another application with a specific id
+    /// @param[in] id identifies the segment
+    /// @param[in] ptr is the start pointer of the segment
+    /// @param[in] size is the size of the segment
+    /// @return true if the registration was successful, otherwise false
+    bool registerPtr(id_t id, ptr_t ptr, uint64_t size) noexcept;
+
+    /// @brief registers the start pointer of a segment with a specific size
+    /// @param[in] ptr is the start pointer of the segment
+    /// @param[in] size is the size of the segment
+    /// @return the id that identifies the segment
+    id_t registerPtr(const ptr_t ptr, uint64_t size = 0U) noexcept;
+
+    /// @brief unregisters the id
+    /// @param[in] id is the id to be unregistered
+    /// @return true if successful, otherwise false
+    /// @attention the relative pointers corresponding to this id become unsafe to use
+    bool unregisterPtr(id_t id) noexcept;
+
+    /// @brief unregisters all ids and also invalidates all relative pointers
```

Comment:
it doesn't invalidate the pinters

Reply:
Yes, it cannot (does not know them nor could it keep track of copies etc.). But they become unsafe to use, this is what is meant.